dataline-integrations/singer/postgres/destination/src/test-integration/java/io/dataline/integration_tests/destinations/TestPostgresDestination.java

```diff
@@ -104,24 +117,49 @@ public void runTest() throws IOException, InterruptedException, SQLException {
     assertLinesMatch(expectedList, actualList);
   }
 
+  @Test
+  public void testConnectionSuccessful() throws InvalidCredentialsException, InvalidCatalogException {
+    SingerCheckConnectionWorker checkConnectionWorker = new SingerCheckConnectionWorker(new SingerDiscoverSchemaWorker(IMAGE_NAME, pbf));
+    StandardCheckConnectionInput inputConfig = new StandardCheckConnectionInput().withConnectionConfiguration(getDbConfig());
+    OutputAndStatus<StandardCheckConnectionOutput> run = checkConnectionWorker.run(inputConfig, jobRoot);
+    assertEquals(SUCCESSFUL, run.getStatus());
+    assertTrue(run.getOutput().isPresent());
+    assertEquals(StandardCheckConnectionOutput.Status.SUCCESS, run.getOutput().get().getStatus());
+  }
+
+  @Test
+  public void testConnectionUnsuccessfulInvalidCreds() throws InvalidCredentialsException, InvalidCatalogException {
+    SingerCheckConnectionWorker checkConnectionWorker = new SingerCheckConnectionWorker(new SingerDiscoverSchemaWorker(IMAGE_NAME, pbf));
+    ObjectNode dbConfig = getDbConfig();
+    dbConfig.put("postgres_password", "superfakepassword_nowaythisworks");
+    StandardCheckConnectionInput inputConfig = new StandardCheckConnectionInput().withConnectionConfiguration(dbConfig);
+
+    OutputAndStatus<StandardCheckConnectionOutput> run = checkConnectionWorker.run(inputConfig, jobRoot);
+    assertEquals(FAILED, run.getStatus());
+    assertTrue(run.getOutput().isPresent());
+    assertEquals(StandardCheckConnectionOutput.Status.FAILURE, run.getOutput().get().getStatus());
+  }
+
   private Process startTarget() throws IOException {
-    return pbf.create(
-        jobRoot, "dataline/integration-singer-postgres-destination", "--config", "config.json")
+    return pbf.create(jobRoot, IMAGE_NAME, "--config", CONFIG_FILENAME)
         .redirectOutput(ProcessBuilder.Redirect.INHERIT)
         .redirectError(ProcessBuilder.Redirect.INHERIT)
         .start();
   }
 
-  private void writeConfigFileToJobRoot() throws IOException {
+  private ObjectNode getDbConfig() {
     Map<String, Object> fullConfig = new HashMap<>();
 
     fullConfig.put("postgres_host", PSQL.getHost());
     fullConfig.put("postgres_port", PSQL.getMappedPort(PostgreSQLContainer.POSTGRESQL_PORT));
     fullConfig.put("postgres_username", PSQL.getUsername());
     fullConfig.put("postgres_password", PSQL.getPassword());
     fullConfig.put("postgres_database", PSQL.getDatabaseName());
+    return (ObjectNode) Jsons.jsonNode(fullConfig);
```

Comment:
You should have a default HashMap config that you clone for every tests and that you modify before making it json instead of exposing the ObjectNode

Reply:
Good point, returning `Map` from this method instead